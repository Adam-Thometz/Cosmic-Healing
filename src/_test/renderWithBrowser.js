import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

function withProvider({ children, initialRoutes }) {
  return (
    <MemoryRouter initialEntries={initialRoutes}>
      {children}
    </MemoryRouter>
)
};

function renderWithBrowser(component, options) {
  const initialRoutes = (options && options.initialRoutes)
    ? options.initialRoutes
    : ['/'];
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      initialRoutes
    }), ...options
  });
};

export default renderWithBrowser;