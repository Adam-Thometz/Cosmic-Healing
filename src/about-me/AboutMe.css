.AboutMe {
  position: relative;
  font-family: Nuckle;
  font-size: 18px;
  font-weight: 600;
}

.AboutMe-img-wrapper {
  width: 40%;
  margin: 0 auto;
}

.AboutMe-img-wrapper img {
  width: 100%;
}

.AboutMe-about {
  display: flex;
  margin: var(--margin) auto;
}

.AboutMe-about-img {
  width: 40%;
}

.AboutMe-about-img img {
  width: 100%;
}

.AboutMe-about-text {
  width: 60%;
  text-align: left;
  margin: 0 150px;
}

.AboutMe-quote {
  color: #828282;
  font-family: Nimbus;
  font-size: 36px;
  width: 70%;
  margin: var(--margin) auto;
}

.AboutMe-specialties {
  display: flex;
  justify-content: space-between;
  margin: var(--margin) 0;
}

.AboutMe-specialties-wrapper {
  text-align: left;
  width: 60%;
  margin: 0 auto;
  padding-left: 5%;
}

.AboutMe-specialties-list {
  margin: 20px;
  padding: 0;
  padding-left: -16px;
}

.AboutMe-specialties li {
  padding-left: 40px;
  margin: 4px 0;
}

.AboutMe-specialties li::marker {
  content: url(../_media/about-me/star-bullet.png);
}

.AboutMe-specialties-img {
  width: 40%;
}

.AboutMe-specialties-img img {
  width: 100%;
}

@media (max-width: 900px) {
  .AboutMe-about {
    flex-direction: column;
    width: 95%;
    margin: 0 auto;
  }

  .AboutMe-about-img {
    width: 80%;
    transform: translateX(-40%);
  }

  .AboutMe-about-text {
    width: 95%;
    margin: 0 auto;
  }

  .AboutMe-quote {
    width: 95%;
  }

  .AboutMe-specialties {
    flex-direction: column-reverse;
    align-items: center;
    width: 95%;
  }

  .AboutMe-specialties-wrapper {
    width: 95%;
  }

  .AboutMe-specialties-img {
    width: 80%;
    margin: var(--margin) 0;
  }
}