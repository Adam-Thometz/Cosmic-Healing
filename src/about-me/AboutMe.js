import React from 'react';

import './AboutMe.css';

import SectionHeader from '../_components/section-header/SectionHeader';
import SmallCaption from '../_components/small-caption/SmallCaption';

import miriam from '../_media/miriam.png'
import hand1 from '../_media/about-me/hand-puzzle1.png';
import hand2 from '../_media/about-me/hand-puzzle2.png';
import Link from '../_components/link/Link';

const AboutMe = () => {
  return (
    <main className='AboutMe'>
      <div className='AboutMe-img-wrapper'>
        <img src={miriam} alt='' />
      </div>
      <SmallCaption text='Meet the Licenced Professional' />
      <SectionHeader text='Hello, my name is Miriam!' />
      <section className='AboutMe-about'>
        <aside className='AboutMe-about-img'>
          <img src={hand1} alt='' />
        </aside>
        <article className='AboutMe-about-text'>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fermentum luctus nisl tortor eu nec malesuada varius vestibulum vel. Rutrum leo elementum, in a nunc gravida. Sit cursus semper id massa a vitae ut lectus. Hac tortor ut pellentesque rhoncus pulvinar integer ipsum. In lorem amet sed arcu quis lacinia sagittis viverra parturient.
          <br />
          <br />
          Placerat dictum montes, sociis tristique enim, sapien nisl dui faucibus. Porttitor faucibus nisi, urna amet, morbi ultrices erat orci nibh. Venenatis, cursus sit sed cursus eu id sit. Sagittis aliquet sit odio vestibulum diam venenatis sed. Turpis penatibus nunc ornare vel aliquam mattis volutpat fermentum risus. Egestas gravida feugiat tellus aliquet turpis pellentesque id pellentesque. Vel dignissim faucibus orci, et magnis in pharetra aenean. Tempus massa et urna, sagittis lacus, sit. Suspendisse mauris, leo ornare tellus commodo vitae.
          <br />
          <br />
          Miriam Nkrumah has trained and worked in various settings, including schools, medical centers and hospitals.
        </article>
      </section>
      <blockquote className='AboutMe-quote'>“I offer sessions designed to help you shift subconscious imprinting, uncover your authentic self, align with your purpose, and live in the present moment.”</blockquote>
      <section className='AboutMe-specialties'>
        <div className='AboutMe-specialties-wrapper'>
          <SmallCaption text='Journey Within' />
          <SectionHeader text='Therapy Specialty' />
          <ul className='AboutMe-specialties-list'>
            <li>Depression</li>
            <li>Grief and Loss</li>
            <li>Life Transitions</li>
            <li>Personal Growth and Self-Esteem</li>
            <li>General relationship challenges (family, friends, co-workers)</li>
          </ul>
          <Link primary ariaLabel='Click to book a free session'>Book a free session</Link>
        </div>
        <aside className='AboutMe-specialties-img'>
          <img src={hand2} alt='' />
        </aside>
      </section>
    </main>
  );
};

export default AboutMe;