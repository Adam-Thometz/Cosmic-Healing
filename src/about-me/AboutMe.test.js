import React from "react";
import renderWithBrowser from '../_test/renderWithBrowser';
import AboutMe from './AboutMe';

describe('AboutMe component', () => {
  it('renders without crashing', () => {
    renderWithBrowser(<AboutMe />);
  });

  it('matches the snapshot', () => {
    const { asFragment } = renderWithBrowser(<AboutMe />);
    expect(asFragment()).toMatchSnapshot();
  });
});