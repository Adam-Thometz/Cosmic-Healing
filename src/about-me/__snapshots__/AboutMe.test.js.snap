// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutMe component matches the snapshot 1`] = `
<DocumentFragment>
  <main
    class="AboutMe"
  >
    <div
      class="AboutMe-img-wrapper"
    >
      <img
        alt=""
        src="miriam.png"
      />
    </div>
    <small
      class="SmallCaption"
      style="color: black;"
    >
      Meet the Licenced Professional
    </small>
    <h2
      class="SectionHeader"
    >
      Hello, my name is Miriam!
    </h2>
    <section
      class="AboutMe-about"
    >
      <aside
        class="AboutMe-about-img"
      >
        <img
          alt=""
          src="hand-puzzle1.png"
        />
      </aside>
      <article
        class="AboutMe-about-text"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fermentum luctus nisl tortor eu nec malesuada varius vestibulum vel. Rutrum leo elementum, in a nunc gravida. Sit cursus semper id massa a vitae ut lectus. Hac tortor ut pellentesque rhoncus pulvinar integer ipsum. In lorem amet sed arcu quis lacinia sagittis viverra parturient.
        <br />
        <br />
        Placerat dictum montes, sociis tristique enim, sapien nisl dui faucibus. Porttitor faucibus nisi, urna amet, morbi ultrices erat orci nibh. Venenatis, cursus sit sed cursus eu id sit. Sagittis aliquet sit odio vestibulum diam venenatis sed. Turpis penatibus nunc ornare vel aliquam mattis volutpat fermentum risus. Egestas gravida feugiat tellus aliquet turpis pellentesque id pellentesque. Vel dignissim faucibus orci, et magnis in pharetra aenean. Tempus massa et urna, sagittis lacus, sit. Suspendisse mauris, leo ornare tellus commodo vitae.
        <br />
        <br />
        Miriam Nkrumah has trained and worked in various settings, including schools, medical centers and hospitals.
      </article>
    </section>
    <blockquote
      class="AboutMe-quote"
    >
      “I offer sessions designed to help you shift subconscious imprinting, uncover your authentic self, align with your purpose, and live in the present moment.”
    </blockquote>
    <section
      class="AboutMe-specialties"
    >
      <div
        class="AboutMe-specialties-wrapper"
      >
        <small
          class="SmallCaption"
          style="color: black;"
        >
          Journey Within
        </small>
        <h2
          class="SectionHeader"
        >
          Therapy Specialty
        </h2>
        <ul
          class="AboutMe-specialties-list"
        >
          <li>
            Depression
          </li>
          <li>
            Grief and Loss
          </li>
          <li>
            Life Transitions
          </li>
          <li>
            Personal Growth and Self-Esteem
          </li>
          <li>
            General relationship challenges (family, friends, co-workers)
          </li>
        </ul>
        <a
          class="Link primary"
        >
          Book a free session
        </a>
      </div>
      <aside
        class="AboutMe-specialties-img"
      >
        <img
          alt=""
          src="hand-puzzle2.png"
        />
      </aside>
    </section>
  </main>
</DocumentFragment>
`;
